src/python/automation/scaling/rupture_set_task_factory.py

```diff
@@ -0,0 +1,94 @@
+#!python3
+
+"""
+Wrapper script that produces a job that can be run either locally or
+to a cluster using PBS
+
+The job is responsible for
+
+ - launching the java application (with its gateway service configured)
+ - executing the python client script + config that calls the Java application
+ - updating the task status, files etc via the toshi_api
+ - shutting down the java app
+
+ The job is either a bash script (for local machine) or
+ a PBS script for the cluster environment
+"""
+import os
+import json
+import scaling.rupture_set_builder_task
+
+class RuptureSetTaskFactory():
+
+    def __init__(self, root_path, working_path, jre_path=None, app_jar_path=None, task_config_path=None, pbs_script=False, initial_gateway_port=25333, pbs_ppn=8, pbs_wall_hours=24):
+        """
+        pbs_script: boolean is this a PBS job?
+        initial_gateway_port: what port to start incrementing from
+        """
+        self._next_port = initial_gateway_port
+        self._pbs_script = pbs_script
+        self._pbs_ppn = pbs_ppn #define hows many processors the PBS job should 'see'
+        self._pbs_nodes = 1 #always ust one PBS node (and which one we don't know)
+        self._pbs_wall_hours = pbs_wall_hours #defines maximum time the jobs is allocated by PBS
+
+        self._jre_path = jre_path or "/opt/sw/java/java-se-8u41-ri/bin"
```

Comment:
jdk11

Reply:
oh wow - hadn't noticed that ! Well spotted thanks!